upload.php

```diff
@@ -222,17 +222,16 @@ function emr_normalize_file_urls( $old, $new ) {
 
 if (is_uploaded_file($_FILES["userfile"]["tmp_name"])) {
 
-	// New method for validating that the uploaded file is allowed, using WP:s internal wp_check_filetype_and_ext() function.
-	$filedata = wp_check_filetype_and_ext($_FILES["userfile"]["tmp_name"], $_FILES["userfile"]["name"]);
+	$filedata = wp_check_filetype( $_FILES["userfile"]["name"] );
 
-	if ($filedata["ext"] == "") {
+	if ( empty( $filedata["ext"] ) || !array_key_exists( $filedata["ext"], get_allowed_mime_types() ) ) {
 		echo esc_html__("File type does not meet security guidelines. Try another.", 'enable-media-replace');
 		exit;
 	}
 
 	$new_filename = $_FILES["userfile"]["name"];
 	$new_filesize = $_FILES["userfile"]["size"];
-	$new_filetype = $filedata["type"];
+	$new_filetype = $filedata[ "ext" ];
```

Comment:
Thanks for your contribution. Why did you change this line? It looks that the $new_filetype needs to be a mime-type (for example image/png) further on and not just the (png) extension, but maybe I'm wrong.

Reply:
(my bad) Good question! lol i think its for the condition under this line. I made this quick update for our customer and everthing work. But i think you should validate the condition after and maybe rename variable.